features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx

```diff
@@ -146,45 +154,29 @@ const APIDetailsTopMenu = (props) => {
         });
     }
 
-    // React.useEffect(() => {
-    //     const restApi = new API();
-    //     const restApiProduct = new APIProduct();
-    //     let apiId = null;
-    //     if (!isAPIProduct) {
-    //         apiId = api.isRevision ? api.revisionedApiId : api.id;
-    //         restApi.getRevisions(apiId).then((response) => {
-    //             setRevision(response.body.list);
-    //         })
-    //             .catch((errorMessage) => {
-    //                 console.error(errorMessage);
-    //                 Alert.error(JSON.stringify(errorMessage));
-    //             });
-    //     } else {
-    //         apiId = api.isRevision ? api.revisionedApiProductId : api.id;
-    //         restApiProduct.getProductRevisions(apiId).then((response) => {
-    //             setRevision(response.body.list);
-    //         })
-    //             .catch((errorMessage) => {
-    //                 console.error(errorMessage);
-    //                 Alert.error(JSON.stringify(errorMessage));
-    //             });
-    //     }
-    // }, []);
-
     const handleChange = (event) => {
         setRevisionId(event.target.value);
     };
 
     const isDownloadable = [API.CONSTS.API, API.CONSTS.APIProduct].includes(api.apiType);
     const { settings, user } = useAppContext();
-    const { allRevisions } = useRevisionContext();
+    const { allRevisions, allEnvRevision } = useRevisionContext();
     const { tenantList } = useContext(ApiContext);
     const userNameSplit = user.name.split('@');
     const tenantDomain = userNameSplit[userNameSplit.length - 1];
     let devportalUrl = `${settings.devportalUrl}/apis/${api.id}/overview`;
     if (tenantList && tenantList.length > 0) {
         devportalUrl = `${settings.devportalUrl}/apis/${api.id}/overview?tenant=${tenantDomain}`;
     }
+
+    function getDeployments(revisionKey) {
+        const array = [];
+        allEnvRevision.filter(
```

Comment:
Shall we use Array.find instead of filter, and what if there  are no matching 

```
env.id === revisionKey
```

Reply:
This function only trigger if there's a match, This is to get the deployment names if there are deployments availble